### PR TITLE
nearai agent dev command starts local UI

### PR DESCRIPTION
**Pull request by gabehamilton**

### CODE REVIEW

**Comment by zavodil** [comment]
I like this idea!

Shall we try to open http://localhost:3000 automatically?

And the choice of name for this command is not entirely clear. Previously, for `nearai agent`, we specified the agent's name (or path), but here we don't do that. Additionally, the command `dev` doesn't seem to associate with launching any kind of UI for me. Perhaps something like `nearai hub` or `nearai ui` would be more intuitive to me.

File: nearai/cli.py [hunk 1]
```diff
@@ -474,6 +475,15 @@ def read_solutions(self, entry: str, status: Optional[bool] = None, verbose: boo
 
 
 class AgentCli:
+    def dev(self) -> None:
+        if not os.path.exists("hub/demo/.env"):
+            shutil.copy("hub/demo/.env.example", "hub/demo/.env")
+
+        ret_val = os.system("npm run dev --prefix hub/demo")
```
Comment (calebjacob) [inline nearai/cli.py:482]: Can this simply be `npm run dev`? I'm not sure what `--prefix hub/demo` is adding here unless I'm totally missing something.

We also need to run `npm i` to install dependencies. Maybe we change to `ret_val = os.system("npm i && npm run dev")`.

Also, the UI requires a minimum of Node 18 - might be worth mentioning somewhere.

I'm not sure how the Python registry works for `pip install nearai`. We'll want to make sure all of `hub/demo`'s source code is published and `nearai dev` works as expected when installed that way.

**Review by calebjacob — CHANGES_REQUESTED** [review]
Nice work! Left one comment to work through.

**Review by calebjacob — APPROVED** [review]
Looks good to start with! We can keep making this more robust with future PRs.

**Pull request merged**